youtube_dl/extractor/viu.py

```diff
@@ -223,20 +269,64 @@ def _real_extract(self, url):
         if not video_data:
             raise ExtractorError('This video is not available in your region.', expected=True)
 
-        stream_data = self._download_json(
-            'https://d1k2us671qcoau.cloudfront.net/distribute_web_%s.php' % country_code,
-            video_id, 'Downloading stream info', query={
-                'ccs_product_id': video_data['ccs_product_id'],
-            }, headers={
-                'Referer': url,
-                'Origin': re.search(r'https?://[^/]+', url).group(0),
-            })['data']['stream']
+        duration_limit = False
+        query = {
+            'ccs_product_id': video_data['ccs_product_id'],
+            'language_flag_id': self._LANGUAGE_FLAG.get(lang_code.lower()) or '3',
+        }
+        headers = {
+            'Referer': re.search(r'https?://[^/]+', url).group(0),
+            'Origin': re.search(r'https?://[^/]+', url).group(0),
```

Comment:
So `Referer` and `Origin` are the same? Why is `url` not valid/possible to use?

Reply:
i just reproduced exactly what the browser sent out. haven't checked if it will work when you pass the full url.